inc/plugins/class-dynamic-content.php

```diff
@@ -45,6 +46,23 @@ public function apply_dynamic_content( $content ) {
 		return preg_replace_callback( $re, array( $this, 'apply_data' ), $content );
 	}
 
+	/**
+	 * Filter post content for dynamic content.
+	 *
+	 * @param string $content Post content.
+	 *
+	 * @return string
+	 */
+	public function apply_dynamic_link( $content ) { 
+		if ( false === strpos( $content, '<o-dynamic-link' ) ) {
+			return $content;
+		}
+
+		$re = '/<o-dynamic-link(?:\s+(?:data-type=["\'](?P<type>[^"\'<>]+)["\']|data-target=["\'](?P<target>[^"\'<>]+)["\']|data-meta-key=["\'](?P<metaKey>[^"\'<>]+)["\']|[a-zA-Z-]+=["\'][^"\'<>]+["\']))*\s*>(?<text>[^ $].*?)<\s*\/\s*o-dynamic-link>/';
```

Comment:
Do you have a tool for making those longs regexp, or just pure pain?

Reply:
It's not that big. It's a lot of repeated part. You can copy in Regex101 and see.